data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,236 @@
+        if [[ -n $sub_pid ]];then
+            kill -9 $sub_pid >> /dev/null 2>&1;sync
+        else
+            break
+        fi
+    done
+}
+
+function end_monitor() {
+
+    #end of monitor for guest console
+    if [ -f ${MONITOR_LOCKED_FILE} ];then
+        echo "End of Monitor for guest console"	
+        #UNLOCK Monitor for guest console	    
+        > ${MONITOR_LOCKED_FILE} ; rm -rf ${MONITOR_LOCKED_FILE}
+        #Kill Pids from Monitor for guest console
+        close_console
+    fi
+}
+
+function start_monitor() {
+
+    #start monitor for guest console
+    get_vm_guestnames_state_running=`virsh list --state-running | grep "${vm_guestnames_types}" | awk '{print $2}'`
+    vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})
+
+    while :
+    do
+        if [ -f ${MONITOR_LOCKED_FILE} ];then
+            #Check with LOCK-UP status
+            cat ${MONITOR_LOCKED_FILE} | grep "LOCKED"
+            if [ $? -eq 0 ];then
+                #KEEP LOCK-UP status
+                #To let just only one virsh console with one vm guest as backgroup PID
+
+                #wait for checking background PID of virsh console
+                sleep 3
+		#Check with background PID of virsh console
+		for vmguest in ${vm_guestnames_array[@]};do
+		    virsh list --state-running | grep ${vmguest}
+                    if [[ $? -eq 0 ]];then
+                        ps aux | grep "virsh console ${vmguest}" | grep -v grep
+                        if [[ $? -ne 0 ]];then
+                            get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+                        fi
+                    else
+                        virsh list --inactive | grep ${vmguest} | grep -v grep
+                        if [[ $? -eq 0 ]];then
+                            echo "${vmguest} is not running now"
+                            echo "No any output from virsh console ${vmguest}"
+                        fi
+                    fi
+                done
+            else
+                #Make sure UNLOCKED
+                echo "UNLOCK Monitor for $(basename $0)!"
+                exit			        
+            fi
+        else
+            echo "Create LOCK-UP for $(basename $0)"
+            echo "LOCKED" > ${MONITOR_LOCKED_FILE}
+            open_console
+        fi
+    done
+}
+
+function check_guest() {
+
+    #check with guest status
+    for vmguest in ${vm_guestnames_array[@]};do
+        echo -e ${vm_guestnames_inactive_array[*]} | grep ${vmguest} >> /dev/null 2>&1
+        if [[ $? -eq 0 ]];then
+            virsh start ${vmguest}
+            vmguest_failed=$((${vmguest_failed} | $(echo $?)))
+
+            #Quit if at least one vm guest failed to start up as normal
+            if [[ ${vmguest_failed} -ne 0 ]];then
+                echo "Fail to boot up ${vmguest} as normal. Please investigate.\n"
+                exit 1
+            fi
+        fi
+    done
+}
```

Comment:
I think "virsh reboot" can be placed here in this check_guest function.

Reply:
imho, I do not agree to add thie **virsh reboot** to this check_guest function.

Reply:
So you wont reboot any running virtual machine ?

Reply:
As mentioned, this script just only handle guest serial console and collect the output from guest serial console without the other operations for vm guest (included virsh reboot). 

So, this script will do not any operations for existed vm guest(s) on vm host - included reboot any running vm guest here.

Reply:
OK. Agree.